sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -150,9 +150,27 @@ object TypeCoercion {
   }
 
   private def findWiderCommonType(types: Seq[DataType]): Option[DataType] = {
+    var awaitingString = false
```

Comment:
Maybe we should make a pass of all dataTypes here, and see whether there exists `StringType` and all other dataTypes can be promoted to `StringType`, under that case we can default return `StringType` instead of `None`, WDYT?